compiler/package.go

```diff
@@ -164,7 +164,10 @@ func Compile(importPath string, files []*ast.File, fileSet *token.FileSet, impor
 	}
 	importContext.Packages[importPath] = typesPkg
 
-	exportData := gcimporter.BExportData(nil, typesPkg)
+	exportData := new(bytes.Buffer)
+	if err := gcexportdata.Write(exportData, nil, typesPkg); err != nil {
+		return nil, fmt.Errorf("failed to write export data: %v", err)
+	}
 	encodedFileSet := bytes.NewBuffer(nil)
```

Comment:
This line uses a very outdated style of initializing a new `bytes.Buffer`, no one writes it like that in anymore. [`bytes.NewBuffer`](https://godoc.org/bytes#NewBuffer) documentation hints at this:

> In most cases, new(Buffer) (or just declaring a Buffer variable) is sufficient to initialize a Buffer.

It's not surprising because this line was written 4 years ago.

I know it's not your code, but it's the line directly following your code, and I think it's not nice how just 4 lines above it is a different style is used to do the same thing:

```Go
exportData := new(bytes.Buffer)
// ... 3 lines
encodedFileSet := bytes.NewBuffer(nil)
```

Let's make this better by updating it so they both use the same style.

I would slightly prefer the `var buf bytes.Buffer` (and using `&` operator) style over `buf := new(bytes.Buffer)`:

```Go
var exportData bytes.Buffer
if err := gcexportdata.Write(&exportData, nil, typesPkg); err != nil {
	return nil, fmt.Errorf("failed to write export data: %v", err)
}
var encodedFileSet bytes.Buffer
if err := fileSet.Write(json.NewEncoder(&encodedFileSet).Encode); err != nil {
```

But if you feel strongly about using `new(bytes.Buffer)` for both variables (and not using `&` operator), I'm okay with that too.

Also, I know this file has yet another instance of `bytes.NewBuffer(nil)`, but it's fine to leave that alone, it's too far from the affected code and hence out of scope of this PR.

Reply:
Done.